### PR TITLE
certbot: add page

### PR DESCRIPTION
**Pull request by sbrl**
----

- [x] The page (if new), does not already exist in the repo.

- [x] The page (if new), has been added to the correct platform folder:  
      `common/` if it's common to all platforms, `linux/` if it's Linux-specific, and so on.

- [x] The page has 8 or fewer examples.

- [x] The PR is appropriately titled:  
      `<command name>: add page` for new pages, or `<command name>: <description of changes>` for pages being edited.

- [x] The page follows the [contributing](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md) guidelines.

### CODE REVIEW

File: pages/linux/letsencrypt.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# letsencrypt
+
+> The Let's Encrypt Agent for automatically obtaining and renewing TLS certificates.
+
+- Obtain a new certificate via webroot authorization:
+
+`sudo letsencrypt --webroot --webroot-path {{path/to/webroot}} --domain {{subdomain.example.com}}`
+
+- Oobtain an untrusted test certificate instead:
```
Comment (pxgamer) [inline pages/linux/letsencrypt.md]: Minor typo here with `Obtain`.

**Comment by agnivade** [comment]
Hmm, I am a bit confused. I thought people used certbot ? There is a CLI for letsencrypt also ? Or are both the same ?

**Comment by pxgamer** [comment]
I was actually thinking of Certbot. 🤔 Looking through some repos, this may be the predecessor to Certbot.

**Comment by sbrl** [comment]
Huh :thinking: I thought that `letsencrypt` was the new one, @agnivade? Can't remember. It works for me :P

@pxgamer: Fixed - thanks for spotting that :-)

**Comment by agnivade** [comment]
Ah, found it here https://github.com/certbot/certbot

> Until May 2016, Certbot was named simply letsencrypt or letsencrypt-auto, depending on install method. Instructions on the Internet, and some pieces of the software, may still refer to this older name.

I think it might be better to use `certbot` rather than `letsencrypt`. What do you think ? 

Also, this is a very important tool, so I would like to expand a bit more to cover more usecases. Specifically I would like to cover all these points - 

```
(default) run   Obtain & install a certificate in your current webserver
    certonly        Obtain or renew a certificate, but do not install it
    renew           Renew all previously obtained certificates that are near expiry
   -d DOMAINS       Comma-separated list of domains to obtain a certificate for

  --apache          Use the Apache plugin for authentication & installation
  --standalone      Run a standalone webserver for authentication
  --nginx           Use the Nginx plugin for authentication & installation
  --webroot         Place files in a server's webroot folder for authentication
  --manual          Obtain certificates interactively, or using shell script hooks

   -n               Run non-interactively
  --test-cert       Obtain a test certificate from a staging server
  --dry-run         Test "renew" or "certonly" without saving any certificates to disk
```

Let's start begin with `certonly --webroot` and then add `--apache` and `--nginx`. Then `renew` and then finally move to `dry-run` and `test-cert` at the end.

**Comment by sbrl** [comment]
Are the switches the same as those I've detailed? I don't actually have `certbot` installed on my server, so I can't test it.

Furthermore, I can't test the `--nginx` and `--apache` options, since I've got it set up for `--webroot` - and I don't want it messing with my configuration files :-(

**Comment by agnivade** [comment]
I think it's okay if we do not practically test every command. We can read the documentation and explain each options clearly. That should be good enough. I can help put up a rough draft if you want.

Yes, the switches are the same I believe.

**Comment by sbrl** [comment]
Ah, ok :-)

How's that, @agnivade?

File: pages/linux/certbot.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+- Obtain a new certificate via nginx authorization, installing the new certificate automatically:
+
+`sudo certbot --nginx --domain {{subdomain.example.com}}`
+
+- Obtain a new certificate via apache authorization, installing the new certificate automatically:
+
+`sudo certbot --apache --domain {{subdomain.example.com}}`
+
+- Renew all Let's Encrypt certificates that expire soon (don't forget to restart any servers that use them afterwards):
```
Comment (agnivade) [inline pages/linux/certbot.md]: What is the actual expiry threshold ?

File: pages/linux/certbot.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# certbot
+
+> The Let's Encrypt Agent for automatically obtaining and renewing TLS certificates.
+> Successor to `letsencrypt`.
+
+- Obtain a new certificate via webroot authorization:
```
Comment (agnivade) [inline pages/linux/certbot.md]: We should clarify that it just obtains the cert  but does not install it.

File: pages/linux/certbot.md (hunk 2 above)
Comment (pxgamer) [reply pages/linux/certbot.md]: As far as I know it's 30 days. :thinking:

Comment (agnivade) [reply pages/linux/certbot.md]: Ok, we should just mention the exact days then. Much more helpful IMO.

**Comment by sbrl** [comment]
Great points, @agnivade and @pxgamer! How's that?

**Comment by agnivade** [comment]
Cool. Do you think we should add the `--manual` option too ?

**Comment by sbrl** [comment]
Hrm. I think that might be an advanced usage, since the man page specifically details that it's laborious - it's much better to automate the process I think. If you don't want the certificate installed automatically, I think that the `--webroot` option should cover almost all other use cases.

**Comment by agnivade** [comment]
Ok.

**Comment by pxgamer** [comment]
All looks good to me. :+1:

**Pull request merged**